recipes-kernel/linux/linux-imx_6.6.bb

```diff
@@ -27,13 +27,66 @@ SRC_URI += " \
 #
 # LINUX_VERSION define should match to the kernel version referenced by SRC_URI and
 # should be updated once patchlevel is merged.
-LINUX_VERSION = "6.6.36"
+LINUX_VERSION = "6.6.52"
 
-KBUILD_DEFCONFIG:mx6-generic-bsp = "imx_v7_defconfig"
-KBUILD_DEFCONFIG:mx7-generic-bsp = "imx_v7_defconfig"
-KBUILD_DEFCONFIG:mx8-generic-bsp = "imx_v8_defconfig"
-KBUILD_DEFCONFIG:mx9-generic-bsp = "imx_v8_defconfig"
+KERNEL_CONFIG_COMMAND = "oe_runmake_call -C ${S} CC="${KERNEL_CC}" O=${B} olddefconfig"
 
 DEFAULT_PREFERENCE = "1"
 
+DO_CONFIG_V7_COPY = "no"
```

Comment:
What the reason for all this code make it work completely different from the expected behavior?